Assets/Mirror/Runtime/NetworkManager.cs

```diff
@@ -659,12 +645,7 @@ public virtual void OnServerReady(NetworkConnection conn)
             NetworkServer.SetClientReady(conn);
         }
 
-        public virtual void OnServerAddPlayer(NetworkConnection conn, NetworkMessage extraMessage)
-        {
-            OnServerAddPlayerInternal(conn);
-        }
-
-        public virtual void OnServerAddPlayer(NetworkConnection conn)
+        public virtual void OnServerAddPlayer(NetworkConnection conn, AddPlayerMessage message)
```

Comment:
this PR is more logical than the other. because we do pass conn, msg everywhere else too.
BUT it's not obvious that extra data can be used then.

=> what if we renamed 'message' to 'extraMessage' here? then it will be obvious that it contains extra data, and we still have the more logical implementation of (conn, msg).

Reply:
It is,  but I want to point out that unlike other methods we pass byte[] in the sending side

we call:

```cs
byte[] extraData = ..;

AddPlayer(conn,   extraData);
```

so in this PR,  we send byte[]  but we receive AddPlayerMessage,

Reply:
true.
well, ClientScene always takes some input, then constructs the message internally and then sends it, which is fine.
onserveraddplayer might as well pass byte[], both would work.

although one upside of passing AddPlayerMessage is that if we ever add more data to the message, the users can access it easily without us having to refactor the code to show the hidden message again.

Reply:
ultimately, with this PR they have more control in long run. no hidden magic. just the message that was received with all that comes with it.

Reply:
All right,  so pick one.   Both PRs solve the issue.  Both have pros and cons.  

Note this change will break anyone doing authentication or overriding OnServerAddPlayer for other reasons,  but it is easy to fix. 

Did you want to merge one of these now?  or you want to wait for the dust to settle on the new Message API?

Reply:
let's just get it done

Reply:
@paulpach can we rename 'message' to 'extraMessage' here to make it at least obvious to others?

Reply:
Fixed